app/src/main/java/com/supertokens/session/FrontToken.java

```diff
@@ -131,7 +131,24 @@ public static void setToken(Context context, String frontToken) {
         }
     }
 
-    private static SharedPreferences getSharedPreferences(Context context) {
-        return context.getSharedPreferences(Constants.SHARED_PREFERENCES_KEY, Context.MODE_PRIVATE);
+    public static void setItem(Context context, String frontToken) {
+        // We update the refresh attempt info here as well, since this means that we've updated the session in some way
+        // This could be both by a refresh call or if the access token was updated in a custom endpoint
+        // By saving every time the access token has been updated, we cause an early retry if
+        // another request has failed with a 401 with the previous access token and the token still exists.
+        // Check the start and end of onUnauthorisedResponse
+        // As a side-effect we reload the anti-csrf token to check if it was changed by another tab.
+        Utils.saveLastAccessTokenUpdate(context);
+        if (frontToken.equalsIgnoreCase("remove")) {
+            FrontToken.removeToken(context);
+            return;
+        }
+
+        FrontToken.setFrontToken(context, frontToken);
+    }
+
+    public static boolean doesTokenExist(Context context) {
+        String frontToken = FrontToken.getFrontTokenFromStorage(context);
+        return frontToken != null;
```

Comment:
is this function used anywhere?

Reply:
It is used when reading the localsessionstate from storage